.github/workflows/build-test.yml

```
@@ -4,12 +4,122 @@ name: Build and test
+      - 'devops/**'
   workflow_dispatch:
 
+env:
+  CI: true
+  ENS_REGISTRY_ADDRESS: ${{secrets.ENS_REGISTRY_ADDRESS}}
+  CHAIN_ID: ${{secrets.CHAIN_ID}}
+  CHAIN_NAME: volta
+  PUBLIC_RESOLVER_ADDRESS: ${{secrets.PUBLIC_RESOLVER_ADDRESS}}
+  RESOLVER_V1_ADDRESS: ${{secrets.RESOLVER_V1_ADDRESS}}
+  RESOLVER_V2_ADDRESS: '0xcf72f16Ab886776232bea2fcf3689761a0b74EfE'
+  DOMAIN_NOTIFIER_ADDRESS: ${{secrets.DOMAIN_NOTIFIER_ADDRESS}}
+  ENS_SYNC_INTERVAL_IN_HOURS: ${{secrets.ENS_SYNC_INTERVAL_IN_HOURS}}
+  ENS_SYNC_ENABLED: ${{secrets.ENS_SYNC_ENABLED}}
+  ENS_URL: ${{secrets.ENS_URL}}
+  REDIS_HOST: ${{secrets.REDIS_HOST}}
+  REDIS_PORT: 6379
+  NATS_CLIENTS_URL: ${{secrets.NATS_CLIENTS_URL}}
+  JWT_PUBLIC_KEY_PATH: ${{secrets.JWT_PUBLIC_KEY}}
+  JWT_PRIVATE_KEY_PATH: ${{secrets.JWT_PRIVATE_KEY}}
+  DB_HOST: ${{secrets.DB_HOST}}
+  DB_PORT: ${{secrets.DB_PORT}}
+  DB_USERNAME: ${{secrets.DB_USERNAME}}
+  DB_PASSWORD: ${{secrets.DB_PASSWORD}}
+  DB_NAME: ${{secrets.DB_NAME}}
+  JWT_ACCESS_TOKEN_EXPIRES_IN: ${{secrets.JWT_ACCESS_TOKEN_EXPIRES_IN}}
+  JWT_REFRESH_TOKEN_EXPIRES_IN: ${{secrets.JWT_REFRESH_TOKEN_EXPIRES_IN}}
+  JWT_ACCESS_TOKEN_NAME: ${{secrets.JWT_ACCESS_TOKEN_NAME}}
+  JWT_REFRESH_TOKEN_NAME: ${{secrets.JWT_REFRESH_TOKEN_NAME}}
+  UNIVERSAL_RESOLVER_URL: ${{secrets.UNIVERSAL_RESOLVER_URL}}
+  ROOT_DOMAIN: iam.ewc
+  DID_SYNC_ENABLED: false
+  DIDDOC_SYNC_INTERVAL_IN_HOURS: 21
+  DID_REGISTRY_ADDRESS: ${{secrets.DID_REGISTRY_ADDRESS}}
+  CLAIM_MANAGER_ADDRESS: ${{secrets.CLAIM_MANAGER_ADDRESS}}
+  ASSETS_MANAGER_ADDRESS: ${{secrets.ASSETS_MANAGER_ADDRESS}}
+  ASSETS_SYNC_INTERVAL_IN_HOURS: ${{secrets.ASSETS_SYNC_INTERVAL_IN_HOURS}}
+  ASSETS_SYNC_HISTORY_INTERVAL_IN_HOURS: ${{secrets.ASSETS_SYNC_HISTORY_INTERVAL_IN_HOURS}}
+  ASSETS_SYNC_ENABLED: ${{secrets.ASSETS_SYNC_ENABLED}}
+  IPFS_URL: ${{secrets.IPFS_URL}}
+  NESTJS_PORT: ${{secrets.NESTJS_PORT}}
+  STRATEGY_CACHE_SERVER: ${{secrets.STRATEGY_CACHE_SERVER}}
+  STRATEGY_PRIVATE_KEY: ${{secrets.STRATEGY_PRIVATE_KEY}}
+  STRATEGY_NUM_BLOCKS_BACK: 10
+  STAKING_POOL_FACTORY_ADDRESS: ${{secrets.STAKING_POOL_FACTORY_ADDRESS}}
+  DID_SYNC_MODE_FULL: false
+  ENABLE_AUTH: ${{secrets.ENABLE_AUTH}}
+  STATUS_LIST_DOMAIN: ${{secrets.STATUS_LIST_DOMAIN}}
+  NATS_ENVIRONMENT_NAME: dev
 
 jobs:
-  deployment:
+  install:
+    name: Install dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup app dependencies
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.10.0'
+      - name: Cache npm cache
+        uses: actions/cache@v3
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: |
+            ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-${{ hashFiles('package-lock.json') }}
+            ${{ runner.os }}-npm-
+      - name: Cache node_modules
+        id: cache-npm
```

`cache-node-modules` looks more precise name
```suggestion
        id: cache-node-modules
```

---

Yes, it needs to correspond to the step name. Thanks!